article-analyzer/src/main/java/com/example/demo/service/CategoryService.java

```diff
@@ -0,0 +1,58 @@
+package com.example.demo.service;
+
+import com.example.demo.files.JsonFileReader;
+import com.google.gson.Gson;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.core.io.ClassPathResource;
+import org.springframework.stereotype.Service;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.Set;
+
+@Service
+public class CategoryService {
+
+    private JsonFileReader jsonFileReader;
+
+    private String categoriesFilePath = "article-analyzer/src/main/resources/categories.json";
```

Comment:
private static final (or public if u plan to use it somewhere else)
Also consider moving it into property file and use @value